core/src/test/java/apoc/convert/ConvertJsonTest.java

```diff
@@ -21,20 +23,104 @@
+                (row) -> assertEquals(EXPECTED_AS_PATH_LIST, row.get("path")));
+
+        testCall(db, "RETURN apoc.json.path($json, '$..columns', []) AS path", Map.of("json", JSON),
+                (row) -> assertEquals(EXPECTED_PATH, row.get("path")));
+
+        // -- convert.fromJsonList
+        testCall(db, "RETURN apoc.convert.fromJsonList($json, '$..columns') AS path", Map.of("json", JSON),
+                (row) -> assertEquals(EXPECTED_PATH_WITH_NULLS, row.get("path")));
+
+        testCall(db, "RETURN apoc.convert.fromJsonList($json, '$..columns', ['AS_PATH_LIST']) AS path", Map.of("json", JSON),
+                (row) -> assertEquals(EXPECTED_AS_PATH_LIST, row.get("path")));
+
+        testCall(db, "RETURN apoc.convert.fromJsonList($json, '$..columns', []) AS path", Map.of("json", JSON),
+                (row) -> assertEquals(List.of(EXPECTED_COLUMNS_MAP), row.get("path")));
+
+
+        db.executeTransactionally("CREATE (n:JsonPathNode {prop: $prop})", Map.of("prop", JSON));
+        
+        // -- convert.getJsonProperty
+        testCall(db, "MATCH (n:JsonPathNode) RETURN apoc.convert.getJsonProperty(n, 'prop', '$..columns') AS path", Map.of("json", JSON),
+                (row) -> assertEquals(EXPECTED_PATH_WITH_NULLS, row.get("path")));
+
+        testCall(db, "MATCH (n:JsonPathNode) RETURN apoc.convert.getJsonProperty(n, 'prop', '$..columns', ['AS_PATH_LIST']) AS path", Map.of("json", JSON),
+                (row) -> assertEquals(EXPECTED_AS_PATH_LIST, row.get("path")));
+
+        testCall(db, "MATCH (n:JsonPathNode) RETURN apoc.convert.getJsonProperty(n, 'prop', '$..columns', []) AS path", Map.of("json", JSON),
+                (row) -> assertEquals(List.of(EXPECTED_COLUMNS_MAP), row.get("path")));
+
+        // -- invalid option
+        try {
+            testCall(db, "RETURN apoc.json.path($json, '$..columns', ['INVALID']) AS path", Map.of("json", JSON),
+                    (row) -> fail("Should fail because of invalid pathOptions"));
+        } catch (RuntimeException e) {
+            assertTrue(ExceptionUtils.getStackTrace(e).contains(PATH_OPTIONS_ERROR_MESSAGE));
+        }
+    }
+    
+    @Test
+    public void testJsonPathWithMapFunctions() {
+        // apoc.convert.getJsonPropertyMap and apoc.convert.fromJsonMap must fail with "ALWAYS_RETURN_LIST" because should return a Map.
+        final Map<String, String> expectedMap = Map.of("_id", "772col2");
+        final String expectedError = "com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `java.util.LinkedHashMap<java.lang.Object,java.lang.Object>` out of START_ARRAY token";
+
+        testCall(db, "RETURN apoc.convert.fromJsonMap($json, '$.columns.col2') AS path", Map.of("json", JSON),
+                (row) -> assertEquals(expectedMap, row.get("path")));
+        try {
+            testCall(db, "RETURN apoc.convert.fromJsonMap($json, '$.columns.col2', ['ALWAYS_RETURN_LIST']) AS path", Map.of("json", JSON),
```

Comment:
Could we catch the `ALWAYS_RETURN_LIST` option early before it runs? Then we could give a better error message that this option doesn't make sense with a map.

Reply:
Surely, added in `JsonUtil.parse()`